### PR TITLE
added tsql function and tests

### PR DESCRIPTION
**Pull request by colindotfun**
@machow wasn't sure about the 'private' class / functions, so let me know what the right way to do this is! 

Thanks :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,5 +43,5 @@ Running tests
 pip install -r requirements.txt
 # may need to uncomment line below
 #pip install -e .
-py.test tests
+pytest tests or python -m pytest tests
 ```
```
Comment (colindotfun) [inline README.md]: oops sorry I may have misunderstood this

File: sqlwhat_ext.py [hunk 2]
```diff
@@ -31,3 +32,33 @@ def check_result2(state, col_names = None, sort = False, match = 'exact'):
...
+
+    # empty test
+    test_has_columns(state)
+    # row test
+    test_nrows(state)
+    # column tests
+    child = _sort_columns_indiv(state)
+    for k in sol_res:
+        msg = "Column `{}` in the solution does not have a column with matching values in your results."
+        test_column(child, k, msg, match = 'exact')
```
Comment (machow) [inline sqlwhat_ext.py]: It looks like match was changed from "any" to "exact". This is different from what's being used in the tsql course, which was changed to "any" in response to Graeme's feedback. Are we changing it back? 

Original code: https://github.com/datacamp/sqlwhat/blob/msft-check-result/sqlwhat/checks/check_result.py#L34

File: tests/test_check_result_tsql.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+from sqlwhat.Reporter import Reporter
```
Comment (machow) [inline tests/test_check_result_tsql.py:1]: awww yeeaahhh!

**Review by machow — CHANGES_REQUESTED** [review]

**Review by machow — APPROVED** [review]

**Comment by machow** [comment]
I changed the SCT to use match = 'any', which was what it was set for for the tsql course. Definitely let me know if it should be changed back (or we can also add a `match` argument to `check_result_mssql`, etc..).

I also modified the unit tests a little bit. They looked good, but I figured it might be useful to demonstrate another way pytest lets people run a lot of similar tests.

Once everything looks good to you let me know and I'll merge / deploy!

File: sqlwhat_ext.py (hunk 2 above)
Comment (colindotfun) [reply sqlwhat_ext.py]: @machow oops sorry, this is entirely the product of derp. It should be `'any'`.

File: tests/test_check_result_tsql.py [hunk 4]
```diff
@@ -26,16 +26,21 @@ def test_check_result_tsql_chain():
...
 
-def test_check_result_tsql_match_pass():
-    state = prepare_state({'a': [1,2,3]}, {'a': [1,2,3]})
+@pytest.mark.parametrize('sol_res,stu_res', [        # pass cases where...
+    ( {'a': [1,2,3]}, {'a': [1,2,3]} ),              # identical
+    ( {'a': [1,2,3]}, {'b': [1,2,3]} ),              # diff colnames
+    ( {'a': [1,2,3]}, {'b': [3,2,1]} ),              # diff colnames + reordered
+    ( {'a': [1]}    , {'a': [1], 'b': [2]} )         # extra cols in student result
+    ])
+def test_check_result_tsql_pass(sol_res, stu_res):
```
Comment (colindotfun) [inline tests/test_check_result_tsql.py:35]: very cool

**Comment by colindotfun** [comment]
@machow looks good, thanks! 🎉

**Pull request merged**